lib/interpreters/split_clustering.rb

```diff
@@ -793,13 +793,19 @@ def self.collect_cluster_data(vrp, nb_clusters)
     def self.duplicate_vehicle(vehicle, timewindow, schedule)
       if timewindow.nil?
         (schedule[:start]..[schedule[:end], 6].min).collect{ |day|
```

Comment:
if schedule[:start] is greater than 6 this creates no TW, is that what we want? I feel like this logic needs `% 7`

Reply:
I would make no sense for user to provide a start bigger than 6 but you are right, nothing prevents this case.

Reply:
In fact this is a tricky case. We should rather ensure start is always smaller or equal to 6 (because it would make no sense otherwise). Otherwise dealing with this case is quite hard : 

If end - start > 7 we need to generate vehicles for every index, otherwise we need to know if start and end are in the same week in order to know if we do (start % 7 .. end % 7) or (end % 7 .. start % 7).

Therefore I will just reject case where start > 6 because that makes no sense anyway.

Reply:
Why do you think it doesn't make sense? For example, a user who uses the day of year as their day index might want to use [182..365] as [start..end] to create a plan for the second half of year because all other indices for their vehicles, services etc comes from their ERP as day #.  If we prevent them, we are forcing them to modify all their day indices with their actual horizon[:start] to make the horizon[:start] to be in the range. 

It is not that hard, it would be similar to what we do for the general case:

```
(0..6).collect{ |day|
next unless (schedule[:start]..schedule[:end]).any?{ |day_index| day_index % 7 == day }

// some code
```

Reply:
I think it makes no sense because range_indices are meant to be used by users that do not know exact date of schedule, but they know if it starts on a monday or friday and they know how long their period is.

If they know the date then they can just use range_date. I think that is how it was supposed to be, and the aim is that no user uses range_indices anyway.

(I thought you were suggesting something else, shorter, anyway). It is not hard to implement, but we have this logic in several parts of the code and it needs more lines than just (schedule[:start]..schedule[:end]).

Reply:
If we ignore the TW case here, I don't see why our code should work for [0, 182] but break for [183, 365]. At the end of the day, the latter case needs a preprocessing of `-183` for all indices and both cases are equal. That's why I can't understand why our code cannot handle these otherwise equivalent problems.

Reply:
From my point of you we are getting off topic but we can discuss it on a Gitlab ticket. I agree we can work on making this more generic but at the same time I think we talked about making our customers always use dates so this will not be a problem in fact.

Reply:
I don't remember we talked about that. I remember that we talked about the fact that we don't need both date and indices inside the codebase, and we should develop only with one and chose indices if I am not mistaken. Otherwise, I don't think it makes sense to prevent users from using indices in the future (because we don't know every software that will interface with optim-api).